core/commons/che-core-commons-lang/src/main/java/org/eclipse/che/commons/lang/ZipUtils.java

```diff
@@ -38,91 +37,88 @@
+   * Recursively add the contents of the given file or directory to a {@link ZipOutputStream}
    *
-   * @param output The stream to write the zip contents to.
-   * @param files The files to add to the zip stream.
+   * @param out The zip file to add to
+   * @param f the file or directory to add to
+   * @param rootPath The path the zip entries are relative to. If f is /a/b/c and rootPath is /a,
+   *     the entry corresponding to f will be named b/c. Must be a prefix of f.
    * @throws IOException
    */
-  public static void zipFiles(OutputStream output, File... files) throws IOException {
-    try (ZipOutputStream zipOut = new ZipOutputStream(output)) {
-      for (File f : files) {
-        if (f.isDirectory()) {
-          addDirectoryEntry(zipOut, f.getName());
-          final String parentPath = f.getParentFile().getAbsolutePath();
-          addDirectoryRecursively(zipOut, parentPath, f, IoUtil.ANY_FILTER);
-        } else if (f.isFile()) {
-          addFileEntry(zipOut, f.getName(), f);
-        }
-      }
+  public static void add(ZipOutputStream out, Path f, Path rootPath) throws IOException {
+    if (!f.startsWith(rootPath)) {
+      throw new IllegalArgumentException(
+          "'" + String.valueOf(rootPath) + "' is not a prefix of '" + String.valueOf(f) + "'");
+    }
+    if (Files.isDirectory(f)) {
+      addDirectory(out, f, rootPath);
+    } else {
+      addFileEntry(out, relativePath(rootPath, f), f.toFile());
     }
   }
 
-  private static void addDirectoryRecursively(
-      ZipOutputStream zipOut, String parentPath, File dir, FilenameFilter filter)
-      throws IOException {
-    final int parentPathLength = parentPath.length() + 1;
-    final LinkedList<File> q = new LinkedList<>();
-    q.add(dir);
-    while (!q.isEmpty()) {
-      final File current = q.pop();
-      final File[] list = current.listFiles();
-      if (list != null) {
-        for (File f : list) {
-          if (filter.accept(current, f.getName())) {
-            final String entryName =
-                f.getAbsolutePath().substring(parentPathLength).replace('\\', '/');
-            if (f.isDirectory()) {
-              addDirectoryEntry(zipOut, entryName);
-              q.push(f);
-            } else if (f.isFile()) {
-              addFileEntry(zipOut, entryName, f);
-            }
-          }
-        }
-      }
+  private static void addDirectory(ZipOutputStream out, Path d, Path root) throws IOException {
+    if (!root.equals(d)) {
+      addDirectoryEntry(out, relativePath(root, d));
+    }
+    try {
+      Files.list(d)
+          .forEach(
+              path -> {
+                try {
+                  add(out, path, root);
+                } catch (IOException e) {
+                  throw new ExceptionWrapper(e);
+                }
+              });
+    } catch (ExceptionWrapper e) {
+      throw (IOException) e.getCause();
     }
   }
 
+  private static String relativePath(Path root, Path f) {
```

Comment:
AFAIK these changes won't be approved for the master branch.

Reply:
They just don't accept refactoring in those components.

Reply:
@skabashnyuk Am I right?

Reply:
I'm ok if unit and selenium test will be ok.